examples/snmp/pfsense.js

```diff
@@ -0,0 +1,1144 @@
+        callback([]);
+    }
+    var length = arrayFn.length;
+    var results = new Array(length);
+    var finished = 0;
+    arrayFn.forEach(function (fn, index) {
+        fn(function (result) {
+            results[index] = result;
+            if (++finished == length) {
+                callback(results);
+            }
+        });
+    });
+}
+
+/**
+ * 
+ * @param {[Function]} functions list of functions to execute sequentially
+ * @param {*} callback called when the sequence of functions is done
+ */
+function execute_seq(functions, callback) {
+    var _this = this;
+    var callbackResult = null;
+    function executeNext(functionIndex) {
+        if (functionIndex == functions.length) return callback.apply(_this, [callbackResult]);
+        functions[functionIndex].apply(_this, [callbackResult, function (result) {
+            callbackResult = result;
+            executeNext.apply(_this, [++functionIndex]);
+        }]);
+    }
+    executeNext.apply(_this, [0]);
+}
+
+/**
+ * 
+ * @param {*} oid snmp oid
+ * @returns the result in {@link config_paramters}
+ */
+function get_result(oid) {
+    return config_paramters.filter(function (p) {
+        return p.oid == oid;
+    }).map(function (p) {
+        return p.result;
+    })[0];
+}
+
+/**
+ * 
+ * @param {*} config contains list of monitoring variables and the way to extract them
+ * @param {*} callback called when all the monitoring variables are fetched
+ */
+function execute_config(config, callback) {
+    var order_groups = [];
+    var orders = config
+        .filter(function (p) { return p.order != null; })
+        .map(function (p) { return p.order; })
+        .reduce(function (a, b) {
+            if (a.indexOf(b) < 0) a.push(b);
+            return a;
+        }, []).sort();
+
+    orders.forEach(function (order) {
+        var group = config.filter(function (p) { return p.order == order; });
+        order_groups.push(group);
+    });
+    var final_group = config.filter(function (p) { return p.order == null; });
+    order_groups.push(final_group);
+
+    var ordred_fns = order_groups.map(function (params) {
+        var fns = params.map(function (p) {
+            return function (callback) {
+                execute_seq.apply(p, [p.exec, callback]);
+            };
+        });
+        return function (last, next) {
+            execute_all(fns, next);
+        };
+    });
+    execute_seq(ordred_fns, function () {
+        var result = [];
+        config_paramters.forEach(function (param) {
+            if (Array.isArray(param._var)) {
+                param._var.forEach(function (_var) {
+                    result.push(_var);
+                });
+            } else {
+                result.push(param._var);
+            }
+        });
+        callback(result);
+    });
+}
+
+/**
+* @remote_procedure
+* @label Validate Association
+* @documentation This procedure is used to check if the snmp is working in pfsense server
+*/
+function validate() {
```

Comment:
You must also validate SSH is working considering this driver is using both protocols.

Also, throw the anauthorized error in case of auth failure for ssh